### PR TITLE
[FEATURE #160577475] Users should receive descriptive signup validation errors

### PR DESCRIPTION
**Pull request by codingedward**
**What does this PR do?**
- Adds descriptive validation error messages during sign up.

**Description of Task to be completed**
- Users should receive a descriptive error message when they enter invalid user parameters in an attempt to signup.

Invalid user parameters include:
- Existing user email
- Invalid email
- Password of length less than 8
- Password not being alphanumeric

**What are the relevant pivotal tracker stories?**
[160577475](https://www.pivotaltracker.com/n/projects/2198533/stories/160577475)

**Additional Comment:**
- I worked on this task in collaboration with @gloriaodipo

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -94,5 +94,16 @@ ENV/
...
+# vim temporary files
+.*.swp
+.*.swn
+.*.swo
+
+# vim python config
+.ycm_extra_conf.py
+
+# migrations folder
+migrations/
```
Comment (FawazFarid) [inline .gitignore]: Is there a reason we are ignoring migrations? According to the [Django migrations documentation](https://docs.djangoproject.com/en/2.1/topics/migrations/):

`
The migration files for each app live in a “migrations” directory inside of that app, and are designed to be committed to, and distributed as part of, its codebase.
You should be making them once on your development machine and then running the same migrations on your colleagues’ machines, your staging machines, and eventually your production machines.
`

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+whitenoise = "==4.1"
+wrapt = "==1.10.11"
+pytz = "==2018.5"
+Django = "==2.1.1"
+PyJWT = "==1.6.4"
+
+[dev-packages]
+
+[requires]
+python_version = "3.7"
```
Comment (FawazFarid) [inline Pipfile:30]: Nice work integrating Pipenv 😄

File: authors/apps/authentication/serializers.py [hunk 3]
```diff
@@ -8,22 +9,50 @@
...
         max_length=128,
         min_length=8,
-        write_only=True
+        write_only=True,
+        required=True,
+        error_messages={
+            'required': 'Password is required',
+            'invalid': 'Password must have a number and a letter',
+            'min_length': 'Password must have at least 8 characters',
+            'max_length': 'Password cannot be more than 128 characters'}
```
Comment (FawazFarid) [inline authors/apps/authentication/serializers.py]: Can we have this as: 

```
error_messages={
    'required': 'Password is required',
    'invalid': 'Password must have a number and a letter',
    'min_length': 'Password must have at least 8 characters',
    'max_length': 'Password cannot be more than 128 characters'
}

```
I think it's more readable that way

File: authors/apps/authentication/serializers.py [hunk 4]
```diff
@@ -8,22 +9,50 @@
...
+    )
+
+    # Ensure the username is at least 4 characters long and  unique
+    username = serializers.CharField(
+        min_length=4,
+        required=True,
+        validators=[UniqueValidator(queryset=User.objects.all())],
+        error_messages={
+            'required': 'Username is required',
+            'min_length': 'Username must have at least 4 characters'}
```
Comment (FawazFarid) [inline authors/apps/authentication/serializers.py]: ditto

**Review by FawazFarid — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (codingedward) [reply .gitignore]: Thank you for the correction. We will fix this.

File: Pipfile (hunk 2 above)
Comment (codingedward) [reply Pipfile:30]: 👍

File: authors/apps/authentication/serializers.py (hunk 3 above)
Comment (codingedward) [reply authors/apps/authentication/serializers.py]: Noted.

File: authors/apps/authentication/serializers.py (hunk 4 above)
Comment (codingedward) [reply authors/apps/authentication/serializers.py]: Noted 👍

**Comment by codingedward** [comment]
Hello @FawazFarid, 

We have implemented the changes requested.

**Review by FawazFarid — APPROVED** [review]

**Pull request merged**